File: src/hackerrank/PalindromeIndex.java
package hackerrank;

public class PalindromeIndex {
    static int attempts = 0;
    static int result = -1;

    public static int palindromeIndex(String s) {
        attempts = 0;
        result = -1;
        // Write your code here
        char[] charArray = s.toCharArray();
        palindrome(charArray, 0, charArray.length - 1);

        return result;
    }

    public static boolean palindrome(char[] charArray, int i, int j) {
        if (i > j) return true;
        if (charArray[i] != charArray[j]) {
            if (attempts > 0) return false;
            else {
                attempts++;
                boolean left = palindrome(charArray, i + 1, j);
                boolean right = palindrome(charArray, i, j - 1);
                if (attempts > 0 && left) result = i;
                else if (attempts > 0 && right) result = j;
                return left || right;
            }
        } else return palindrome(charArray, i + 1, j - 1);

    }

}

File: src/hackerrank/PalindromeIndex.kt
package hackerrank

class PalindromeIndex {
    fun palindromeIndex(s: String): Int {
        // Write your code here
        var result = -1
        var i = 0
        var j = s.length - 1
        while (i < j) {
            val isValid = isValid(s, i, j)
            if (s[i] == s[j]) {
                i++
                j--
            } else {
                if (!isValid) return -1
                else {
                  //  isValid = false
                    if (s[i + 1] == s[j]) {
                        result = i
                        i += 2
                        j--
                    } else if (s[i] == s[j - 1]) {
                        result = j
                        i++
                        j -= 2
                    }
                }
            }
        }
        return result
    }

    fun isValid(s: String, i: Int, j: Int): Boolean {
        return (i < j && s[i] == s[j]) || isValid(s, i + 1, j) || isValid(s, i, j - 1)
    }
}